### DISCUSSION TITLE
How to use Dev Containers in VSCode without Docker Desktop?

### DISCUSSION THREAD

**Discussion by ISkomorokh** [devcontainers/community#99] [Q&A] [answered]
Hi all,
Could someone help me set up VSCode Dev Containers with Dockerd running on WSL Ubuntu?

I have the following configuration: 
Host OS - Windows 11
WSL2 - Ubuntu 22.04.3 LTS

To start the Docker Engine, I followed the instructions from this good post https://medium.com/@ferarias/docker-in-windows-11-using-wsl2-8e30faddc32c

With these settings, I manage to successfully run docker containers from my host, but VSCode Dev Containers does not work 😔.

Here's where the whole process breaks down:
![image](https://github.com/devcontainers/community/assets/1455188/e34a0884-30cb-4614-b631-1b0080bd1ae4)

If I install Docker Desktop, everything works. But I can't install it because of policies.

**Comment by eljog** [comment]
Thank you for the question and sharing the details.
@chrmarti do you have any suggestions here?

**Comment by chrmarti** [comment]
Try enabling Dev Containers: Execute in WSL in the VS Code user settings:
<img width="1007" alt="image" src="https://github.com/devcontainers/community/assets/9205389/c616db75-e04b-4535-9098-bb4f9637ae20">

**Reply by ISkomorokh** [reply]
That setting only works for me when I Connect to WSL first and do everything inside, but my question is about working with Dev Containers from the Host OS without connecting to the WSL first.

![image](https://github.com/devcontainers/community/assets/1455188/1d17d468-028c-47c8-8b39-67c0458b6c03)

**Reply by chrmarti** [reply, marked as answer]
This should also work with Windows folders. If you want to use a WSL folder (e.g., for performance) without connecting to WSL first, you could open `\\wsl.localhost\<wsl distro>\<wsl path>` (e.g., `\\wsl.localhost\Ubuntu\home\chrmarti\development\myproject`) and reopen that in a dev container.

**Reply by ISkomorokh** [reply]
@chrmarti, thanks for your help! This way, it worked fine. And even more - now it looks to be working fine as I initially wanted it to.

**Comment by ISkomorokh** [comment]
Today, I discovered that it doesn't really work as intended :( I forgot that I didn't update `hosts` in the docked configuration; thus, `docker` isn't accessible from the host OS directly. It is only accessible via `wsl docker`. This puts limitations on the `Visual Studio 2022` and `VS Code` docker tools as they can't see the docker.
 
Anyway, at least I can use Dev Containers.
